openstack/db/v1/instances/results.go

```diff
@@ -49,13 +61,33 @@ type Instance struct {
 	// The build status of the instance.
 	Status string
 
+	// Fault information (only available when the instance has errored)
+	Fault Fault
```

Comment:
Let's change the type to `*Fault`. That way it can be checked like
```go
if instance.Fault != nil {
...
}
```
Is that how it's expected to be used? (Specifically, I suspect it's the case that `Fault` may be populated when the error returned by Gophercloud from a call to OpenStack is `nil`.)